src/Polytopes/Graphs.jl

```diff
@@ -0,0 +1,659 @@
+
+Add a `n` new vertices to the graph `g`.
+
+# Examples
+```jldoctest
+julia> g = Graph{Directed}(2);
+
+julia> nv(g)
+2
+
+julia> add_vertices!(g, 5);
+
+julia> nv(g)
+7
+```
+"""
+function add_vertices!(g::Graph{T}, n::Int64) where {T <: Union{Directed, Undirected}}
+    for i = 1:n
+        add_vertex!(g)
+    end
+end
+
+
+################################################################################
+################################################################################
+##  Edges
+################################################################################
+################################################################################
+struct Edge
+    source::Int64
+    target::Int64
+end
+
+
+@doc Markdown.doc"""
+    src(e::Edge)
+
+Return the source of an edge.
+
+# Examples
+```jldoctest
+julia> g = complete_graph(2);
+
+julia> E = collect(edges(g));
+
+julia> e = E[1]
+Edge(2, 1)
+
+julia> src(e)
+2
+```
+"""
+function src(e::Edge)
+    return e.source
+end
+
+
+@doc Markdown.doc"""
+    dst(e::Edge)
+
+Return the destination of an edge.
+
+# Examples
+```jldoctest
+julia> g = complete_graph(2);
+
+julia> E = collect(edges(g));
+
+julia> e = E[1]
+Edge(2, 1)
+
+julia> dst(e)
+1
+```
+"""
+function dst(e::Edge)
+    return e.target
+end
+
+
+@doc Markdown.doc"""
+    reverse(e::Edge)
+
+Return the edge in the opposite direction of the edge `e`.
+
+# Examples
+```jldoctest
+julia> g = complete_graph(2);
+
+julia> E = collect(edges(g));
+
+julia> e = E[1]
+Edge(2, 1)
+
+julia> Graphs.reverse(e)
+Edge(1, 2)
+```
+"""
+function reverse(e::Edge)
+    return Edge(dst(e), src(e))
+end
+
+
+struct EdgeIterator
+    pm_itr::Polymake.GraphEdgeIterator{T} where {T <: Union{Directed, Undirected}}
+    l::Int64
+end
+Base.length(eitr::EdgeIterator) = eitr.l
+Base.eltype(::Type{EdgeIterator}) = Edge
+
+function Base.iterate(eitr::EdgeIterator, index = 1)
+    if index > eitr.l
+        return nothing
+    else
+        e = Polymake.get_element(eitr.pm_itr)
+        s = Polymake.first(e)
+        t = Polymake.last(e)
+        edge = Edge(s+1, t+1)
+        Polymake.increment(eitr.pm_itr)
+        return (edge, index+1)
+    end
+end
+
+
+################################################################################
+################################################################################
+##  Accessing properties
+################################################################################
+################################################################################
+@doc Markdown.doc"""
+    nv(g::Graph{T}) where {T <: Union{Directed, Undirected}}
+
+Return the number of vertices of a graph.
+
+# Examples
+The edge graph of the cube has eight vertices, just like the cube itself.
+```jldoctest
+julia> c = cube(3);
+
+julia> g = edgegraph(c);
+
+julia> nv(g)
+8
+```
+"""
+function nv(g::Graph{T}) where {T <: Union{Directed, Undirected}}
```

Comment:
Ugh... `nv`, really? LightGraphs, whyyyyy??? I don't get this desire to shorten identifiers so much, esp. for exported stuff. If everybody does it, composing code from different sources becomes ever harder. Urgh.

And `ne` is even worse, it stands for `not equal` in a bunch of places (TeX, many assembler dialects). 

How about we give it a sane name, e.g `nverts`, and similar `ne` -> `nedges`. And if someone for some reason really needs `LightGraphs.jl` compatibility (is that actually a thing?), then adding alias methods is a matter of adding 2 short lines of code.

Reply:
We agreed above to add those awful names, but hide them in the `Graphs` submodule. The proper names would be `num_edges` I guess (to be in line with the other `num_*` functions)?

Reply:
OK if it is hidden, that's better.

The naming issue is less clear to me. Loading Oscar, I see only a single `num_` function, `num_coeff`. But I see a bunch of `number_` functions (some from GAP/groups, e.g. `number_conjugacy_classes`; others from Polymake/Polytopes, e.g. `number_of_factors`), and of course also `n*` functions (e.g. `nrows`, `ncols`, `nvertices`, `nfacets`); of course Julia itself uses `n*` e.g. `ndims`, `nfields`, ...). Perhaps another issue should be opened to settle *that* bit of (in)consistency... *sigh*

Reply:
Ah, you are right. Sad.